src/rules/selector-attribute-square-brackets-space-inside/index.js

```diff
@@ -0,0 +1,81 @@
+import selectorParser from "postcss-selector-parser"
+import {
+  report,
+  ruleMessages,
+  styleSearch,
+  validateOptions,
+} from "../../utils"
+
+export const ruleName = "selector-attribute-square-brackets-space-inside"
+
+export const messages = ruleMessages(ruleName, {
+  expectedOpening: "Expected single space after \"[\"",
+  rejectedOpening: "Unexpected whitespace after \"[\"",
+  expectedClosing: "Expected single space before \"]\"",
+  rejectedClosing: "Unexpected whitespace before \"]\"",
+})
+
+export default function (expectation) {
+  return (root, result) => {
+    const validOptions = validateOptions(result, ruleName, {
+      actual: expectation,
+      possible: [
+        "always",
+        "never",
+      ],
+    })
+    if (!validOptions) { return }
+
+    root.walkRules(rule => {
+      selectorParser(selectorTree => {
+        selectorTree.eachAttribute(attributeNode => {
+          const attributeSelectorString = attributeNode.toString()
+
+          styleSearch({ source: attributeSelectorString, target: "[" }, match => {
```

Comment:
@davidtheclark 

```css
[ target=blank ] { }
```

```shell
Attribute {
  operator: '=',
  value: 'blank ',
  source: { start: { line: 1, column: 1 }, end: { line: 1, column: 16 } },
  sourceIndex: 0,
  attribute: ' target',
  spaces: { before: '', after: '' },
  type: 'attribute',
  raw: {},
  parent: 
   Selector {
     spaces: { before: '', after: '' },
     nodes: [ [Circular] ],
     type: 'selector',
     parent: 
      Root {
        spaces: [Object],
        nodes: [Object],
        type: 'root',
        lastEach: 1,
        indexes: [Object] },
     lastEach: 1,
     indexes: { '1': 0 } } }
```

Spaces in `attribute` and `value`, it is normal check this in spaces?